src/hbbft.erl

```diff
@@ -24,22 +24,19 @@
 
 -ifdef(TEST).
 -export([
-    encrypt/3,
-    get_encrypted_key/2,
-    decrypt/2,
+    encrypt/2,
     encode_list/1,
     abstraction_breaking_set_acs_results/2,
     abstraction_breaking_set_enc_keys/2
 ]).
 -endif.
 
 -type acs_results() :: [{non_neg_integer(), binary()}].
--type enc_keys() :: #{non_neg_integer() => tc_ciphertext:ciphertext() | tpke_pubkey:ciphertext()}.
+-type enc_keys() :: #{non_neg_integer() => tc_ciphertext:ciphertext()}.
 
 -record(hbbft_data, {
     batch_size :: pos_integer(),
-    curve :: curve(),
-    key_share :: key_share(),
+    key_share :: undefined | tc_key_share:tc_key_share(),
```

Comment:
When can the key share be undefined?

Reply:
@Vagabond Great question. I simply replaced:

```erlang
-type key_share() :: undefined | tc_key_share:tc_key_share() | tpke_privkey:privkey().
...
key_share :: key_share(),
```

with

```erlang
key_share :: undefined | tc_key_share:tc_key_share(),
```

without checking if `undefined` is still needed at all.

Will check.

Reply:
@Vagabond Indeed it doesn't look possible, since `init` requires it and it never appears to be removed. 

Do we (still) need `have_key/1` then?

Reply:
It appears to be used in miner as some extra sanity assertion:
https://github.com/helium/miner/blob/master/src/miner_consensus_mgr.erl#L396
https://github.com/helium/miner/blob/master/src/miner_consensus_mgr.erl#L924
https://github.com/helium/miner/blob/master/src/miner_consensus_mgr.erl#L1041